### ISSUE TITLE
Wait Time for Data Stream

### ISSUE THREAD

**Issue by travis23** [rosenbrockc/live-serial#10]
Currently, the code waits for about 0.5 seconds before determining that there is no data available (and raising an exception). The user ought to be able to specify how long they are willing to wait for the hardware to initialize the stream.

Also, if the timeout is too low on the port configuration, then the data comes muddled (with new lines only showing up every few buffer flushes and the other data mixed up). If the timeout is made longer, then it works great. We either need to automate this, or add it to a wiki somewhere so the user can troubleshoot their session.

If the data is muddled then the code raises `ValueError: Live feed has no sensor data keys. Can't setup plotting.`

**Comment by travis23** [comment]
Example of the timeout problem: `timeout=0.01`:

```
b'1029.62 476.65 \r'
b'31.45 475.28 \r\n'
b'224 1340.84 103'
b'85 1339.47 1028.'
b' 1338.55 1027.7'
b'1339.47 1029.62'
b'39.01 1029.62 4'
b'7.18 1028.25 47'
b'.10 1030.08 476.'
b'2 1029.16 475.7'
b' 1030.99 474.82 '
b'029.16 475.74 \r'
b'30.08 475.28 \r\n'
```

Reduce to `timeout=0.005`:

```
b'.75 997.17 \r\n'
b'229.33 614.66 1'
b''
b''
b'718 1'
b''
b''
b' 1230.24 615.12'
b''
b' \r\n'
```

Increase to `timeout=0.025`:

```
b'1230.70 613.75 998.54 \r\n'
b'24 1230.70 613.75 994.43 \r\n'
b'71 1338.10 1029.62 474.82 \r\n'
b'161 1338.55 1027.34 476.65 \r\n'
b'223 1338.10 1027.79 474.82 \r\n'
```

**Comment by rosenbrockc** [comment]
Available in Revision 0.1.2
